pkg/api/types.go

```diff
@@ -180,6 +180,139 @@ type VolumeSource struct {
 	GitRepo *GitRepoVolumeSource `json:"gitRepo"`
 	// Secret represents a secret that should populate this volume.
 	Secret *SecretVolumeSource `json:"secret"`
+	// PersistentVolumeClaimVolumeSource represents a reference to a PersistentVolumeClaim in the same namespace
+	PersistentVolumeClaimVolumeSource *PersistentVolumeClaimVolumeSource `json:"persistentVolumeClaim,omitempty"`
+}
+
+// Similar to VolumeSource but meant for the administrator who creates PVs.
+type PersistentVolumeSource struct {
+
+	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
+	// kubelet's host machine and then exposed to the pod.
+	GCEPersistentDisk *GCEPersistentDiskVolumeSource `json:"persistentDisk"`
+
+	// AWSElasticBlockStore represents an EBS volume in AWS that is attached to
```

Comment:
After I introduce the interface method for getting AccessModes from the volume itself, I won't need these.